source/common/common/utility.h

```diff
@@ -48,6 +52,13 @@ class DateFormatter {
   const std::string& formatString() const { return format_string_; }
 
 private:
+  std::string fromTime(time_t time, const std::string& new_format_string) const;
+  std::string setCustomField(const std::string& field, const std::string& value) const;
+
+  struct CustomFields {
+    static const std::string& f() { CONSTRUCT_ON_FIRST_USE(std::string, "f"); }
```

Comment:
Would a `ConstSingleton` be simpler here?

Reply:
Done.